### PR TITLE
[prometheus] Add deploymentLabels to prometheus server chart

### LINKED ISSUE #699: [prometheus] deployment labels are not configurable
**Is your feature request related to a problem? Please describe.**
We're currently deploying using the prometheus helm chart, but have a need to configure deployment labels. This is not currently configurable

**Describe the solution you'd like**
I'd like to be able to configure deployment labels the same way deployment annotations are configurable.

**Describe alternatives you've considered**
We've considered downloading a local copy of the chart, but figured this could be a common enough usecase to propose an update.

### PR DESCRIPTION
**Pull request by codeboten**
#### What this PR does / why we need it:
This allows users to configure labels via the `server.deploymentLabels` configuration option

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #699

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

Signed-off-by: Alex Boten <aboten@lightstep.com>

### CODE REVIEW

File: charts/prometheus/templates/server/deploy.yaml [hunk 1]
```diff
@@ -9,6 +9,9 @@ metadata:
 {{- end }}
   labels:
     {{- include "prometheus.server.labels" . | nindent 4 }}
+{{- if .Values.server.deploymentLabels }}
```
Comment (monotek) [inline charts/prometheus/templates/server/deploy.yaml:12]: Imho this should be named extra labels and added to all resources.

**Review by monotek — CHANGES_REQUESTED** [review]

**Comment by monotek** [comment]
Maybe we could even extend the metalabels helper, so it includes some additional labels from the values file?

see: https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus/templates/_helpers.tpl#L24

**Comment by codeboten** [comment]
> Maybe we could even extend the metalabels helper, so it includes some additional labels from the values file?
> 
> see: https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus/templates/_helpers.tpl#L24

Happy to go either ways, could use some guidance using the metalabels helper if that's the preferred approach.

File: charts/prometheus/templates/server/deploy.yaml (hunk 1 above)
Comment (jmacd) [reply charts/prometheus/templates/server/deploy.yaml:12]: Yes, for example the node_exporter daemonset can't be labeled either.

**Review by jmacd — APPROVED** [review]

**Pull request closed without merging**